### PR TITLE
[Ebook settings] Create settings ( issue #14)

### PR DESCRIPTION
**Pull request by kastrubait**
- [x] add ebookSettings component
- [x] initialization service ebookSettings
- [x] add styling to the component
- [ ] configure routing

### CODE REVIEW

File: rslang/src/app/shared/models/optionss.model.ts [hunk 1]
```diff
@@ -0,0 +1,5 @@
+export interface Options {
```
Comment (Anntol) [inline rslang/src/app/shared/models/optionss.model.ts]: typo (2 s) in filename?

Comment (kastrubait) [reply rslang/src/app/shared/models/optionss.model.ts]: This is not my file)) Why do you write comments on someone else's work?

Comment (Anntol) [reply rslang/src/app/shared/models/optionss.model.ts]: It is shown as part of "Files changes" for this PR

File: rslang/src/app/ebook/components/ebook-home/ebook-home.component.ts [hunk 2]
```diff
@@ -1,21 +1,50 @@
...
+    private localStorageService: LocalStorageService,
+  ) {}
 
   ngOnInit(): void {
     this.words = this.wordsDataService.GetWords();
+
+    this.ebookSettingsSubscription = this.ebookSettingsChanges$
+      .subscribe(
+        (events: StorageChanges) => {
+          if (events.type === 'set' && events.key === 'bookSettings') {
```
Comment (Nikitzu) [inline rslang/src/app/ebook/components/ebook-home/ebook-home.component.ts]: Can we use enums for event type and key?

File: rslang/src/app/ebook/components/ebook-settings/ebook-settings.component.ts [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+  templateUrl: './ebook-settings.component.html',
+  styleUrls: ['./ebook-settings.component.scss'],
+})
+export class EbookSettings {
+  isUserAuthenticated = false;
+  @Input() userBookSettings: UserBookSettings;
+  @Output() bookSettingsChanged: EventEmitter<UserBookSettings> = new EventEmitter<UserBookSettings>();
+
+  onSubmit(settingsForm: NgForm): void {
+    if (settingsForm.invalid) {
```
Comment (Nikitzu) [inline rslang/src/app/ebook/components/ebook-settings/ebook-settings.component.ts]: can we disable submit is form is invalid in order not to check it here?

File: rslang/src/app/ebook/components/words-collections/words-collections.component.spec.ts [hunk 4]
```diff
@@ -35,6 +35,6 @@ describe('WordsCollections', () => {
     ];
     comp.wordsCollections = wordsCollections;
 
-    comp.getGroup.subscribe(selectedWordsCollection => expect(selectedWordsCollection).toContain(wordsCollections))
+    comp.groupIdChanged.subscribe(selectedWordsCollection => expect(selectedWordsCollection).toContain(wordsCollections))
```
Comment (Nikitzu) [inline rslang/src/app/ebook/components/words-collections/words-collections.component.spec.ts:38]: please avoid using expect in async actions. It is not a good way of testing

File: rslang/src/app/ebook/services/ebook-settings-init.service.ts [hunk 5]
```diff
@@ -0,0 +1,23 @@
...
+import { LocalStorageService } from '../../shared/services/local-storage.service';
+import { EBOOK_SETTINGS } from '../constants/ebook-settings';
+
+@Injectable({
+  providedIn: 'root',
+})
+export class EbookSettingsInitService {
+  constructor(private localStorageService: LocalStorageService) { }
+
+  Init(): Promise<void> {
```
Comment (Nikitzu) [inline rslang/src/app/ebook/services/ebook-settings-init.service.ts]: Why capital letter for method name?

File: rslang/src/app/ebook/services/ebook-settings-init.service.ts [hunk 6]
```diff
@@ -0,0 +1,23 @@
+import { Injectable } from '@angular/core';
+
+import { LocalStorageService } from '../../shared/services/local-storage.service';
+import { EBOOK_SETTINGS } from '../constants/ebook-settings';
+
+@Injectable({
+  providedIn: 'root',
+})
+export class EbookSettingsInitService {
```
Comment (Nikitzu) [inline rslang/src/app/ebook/services/ebook-settings-init.service.ts]: What does it mean init service? all service are inited

File: rslang/src/app/ebook/services/ebook-settings-init.service.ts [hunk 7]
```diff
@@ -0,0 +1,23 @@
...
+  constructor(private localStorageService: LocalStorageService) { }
+
+  Init(): Promise<void> {
+    return new Promise<void>((resolve) => {
```
Comment (Nikitzu) [inline rslang/src/app/ebook/services/ebook-settings-init.service.ts]: Please if you want async use observables

File: rslang/src/app/shared/models/optionss.model.ts (hunk 1 above)
Comment (Nikitzu) [reply rslang/src/app/shared/models/optionss.model.ts]: As soon as file is changed here. If you made mistake in branches so it is not a problem of reviewer

**Review by Nikitzu — CHANGES_REQUESTED** [review]

File: rslang/src/app/ebook/components/ebook-home/ebook-home.component.ts [hunk 8]
```diff
@@ -1,21 +1,50 @@
...
+            this.userBookSettings = JSON.parse(events.value as string) as UserBookSettings;
+          }
+        },
+      );
+  }
+
+  changeSelectedBookSetting(bookSettingsChanged: UserBookSettings): void {
```
Comment (whattablackhole) [inline rslang/src/app/ebook/components/ebook-home/ebook-home.component.ts:49]: don't mix 'ebook' and 'book' fixture naming

File: rslang/src/app/shared/models/optionss.model.ts (hunk 1 above)
Comment (kastrubait) [reply rslang/src/app/shared/models/optionss.model.ts]: fix

File: rslang/src/app/ebook/services/ebook-settings-init.service.ts (hunk 5 above)
Comment (kastrubait) [reply rslang/src/app/ebook/services/ebook-settings-init.service.ts]: fix

File: rslang/src/app/ebook/components/ebook-home/ebook-home.component.ts (hunk 2 above)
Comment (kastrubait) [reply rslang/src/app/ebook/components/ebook-home/ebook-home.component.ts]: I think it's possible

File: rslang/src/app/ebook/components/words-collections/words-collections.component.spec.ts (hunk 4 above)
Comment (kastrubait) [reply rslang/src/app/ebook/components/words-collections/words-collections.component.spec.ts:38]: And what do they use if not expect ??

File: rslang/src/app/ebook/components/ebook-home/ebook-home.component.ts (hunk 8 above)
Comment (kastrubait) [reply rslang/src/app/ebook/components/ebook-home/ebook-home.component.ts:49]: fix

File: rslang/src/app/ebook/services/ebook-settings-init.service.ts (hunk 6 above)
Comment (kastrubait) [reply rslang/src/app/ebook/services/ebook-settings-init.service.ts]: I planned, when opening the ebook, to initialize the user settings, if the user is not registered, then read the setting from the localstorage or config file.

But Correcting comments- I got even more bugs and got stuck (( I need help

File: rslang/src/app/ebook/components/words-collections/words-collections.component.spec.ts (hunk 4 above)
Comment (Nikitzu) [reply rslang/src/app/ebook/components/words-collections/words-collections.component.spec.ts:38]: You need to write test in way that will not check async action. For example you can check not the value but the call of the subscribe with the help of spy function

File: rslang/src/app/shared/models/change-storage-type.model.ts [hunk 9]
```diff
@@ -0,0 +1,4 @@
+export enum LocalStorageType {
+  set = 'set',
```
Comment (Nikitzu) [inline rslang/src/app/shared/models/change-storage-type.model.ts]: capital letter.

File: rslang/src/app/shared/models/local-storage-keys.model.ts [hunk 10]
```diff
@@ -0,0 +1,6 @@
+export enum LocalStorageKey {
+  ebookSettings = 'ebookSettings',
```
Comment (Nikitzu) [inline rslang/src/app/shared/models/local-storage-keys.model.ts]: capital letter

File: rslang/src/app/ebook/components/ebook-home/ebook-home.component.ts [hunk 11]
```diff
@@ -1,21 +1,53 @@
...
   templateUrl: './ebook-home.component.html',
   providers: [WordsDataService],
 })
-export class EbookHome implements OnInit {
+export class EbookHome implements OnInit, OnDestroy {
   wordsCollections: WordsCollection[] = CONFIG_EBOOK.collections;
   words!: Word[];
+  userBookSettings: UserBookSettings;
+  ebookSettingsSubscription = new Subscription();
+  ebookSettingsChanges$ = this.localStorageService.changes$;
```
Comment (Nikitzu) [inline rslang/src/app/ebook/components/ebook-home/ebook-home.component.ts]: for what do we need this var? cant we doo all the same without it?

File: rslang/src/app/ebook/components/words-collections/words-collections.component.html [hunk 12]
```diff
@@ -0,0 +1,16 @@
+<section class="collections">
+  <ng-container>
+    <div class="collections-container">
+      <div
+        class="collections-container--item"
+        *ngFor="let collection of wordsCollections; index as item"
```
Comment (Nikitzu) [inline rslang/src/app/ebook/components/words-collections/words-collections.component.html]: maybe index as i?

File: rslang/src/app/ebook/ebook.module.ts [hunk 13]
```diff
@@ -1,11 +1,25 @@
...
-
 import { EbookRoutingModule } from './ebook-routing.module';
+
 import { EbookHome } from './components/ebook-home/ebook-home.component';
-import { WordsCollections } from './components/collections/words-collections.component';
+import { WordsCollections } from './components/words-collections/words-collections.component';
+import { EbookSettings } from './components/ebook-settings/ebook-settings.component';
+import { EbookSettingsInitService } from './services/ebook-settings-init.service';
+
+export function initializeEbookSettings(ebookSettingsInitService: EbookSettingsInitService) {
```
Comment (Nikitzu) [inline rslang/src/app/ebook/ebook.module.ts]: What? Cant we use service for settings?

File: rslang/src/app/ebook/services/ebook-settings-init.service.ts [hunk 14]
```diff
@@ -0,0 +1,23 @@
+import { Injectable } from '@angular/core';
+
+import { LocalStorageService } from '../../shared/services/local-storage.service';
+import { LocalStorageKey } from '../../shared/models/local-storage-keys.model';
+import { EBOOK_SETTINGS } from '../constants/ebook-settings';
+import { UserBookSettings } from '../models/user-book-settings.model';
+
+@Injectable({
+  providedIn: 'root',
```
Comment (Nikitzu) [inline rslang/src/app/ebook/services/ebook-settings-init.service.ts]: Why this service is root?

File: rslang/src/app/ebook/services/ebook-settings-init.service.ts (hunk 6 above)
Comment (Nikitzu) [reply rslang/src/app/ebook/services/ebook-settings-init.service.ts]: Your plans don't matter. We should not use INIT in service name as soon as it is too missleading

File: rslang/src/app/ebook/services/ebook-settings-init.service.ts [hunk 15]
```diff
@@ -0,0 +1,23 @@
...
+})
+export class EbookSettingsInitService {
+  constructor(private localStorageService: LocalStorageService) { }
+  ebookSettingsData: UserBookSettings;
+
+  init(): void {
+    const data = this.localStorageService.getItem(LocalStorageKey.ebookSettings);
+    if (data === null) {
```
Comment (Nikitzu) [inline rslang/src/app/ebook/services/ebook-settings-init.service.ts]: maybe if !data?

**Review by Nikitzu — CHANGES_REQUESTED** [review]

File: rslang/src/app/ebook/components/ebook-settings/ebook-settings.component.ts (hunk 3 above)
Comment (kastrubait) [reply rslang/src/app/ebook/components/ebook-settings/ebook-settings.component.ts]: checking the valid before submitting the form

File: rslang/src/app/ebook/services/ebook-settings-init.service.ts (hunk 6 above)
Comment (kastrubait) [reply rslang/src/app/ebook/services/ebook-settings-init.service.ts]: changed the naming of the service

File: rslang/src/app/ebook/services/ebook-settings-init.service.ts (hunk 7 above)
Comment (kastrubait) [reply rslang/src/app/ebook/services/ebook-settings-init.service.ts]: EbookSettingsInitService is temporarily unused

File: rslang/src/app/shared/models/change-storage-type.model.ts (hunk 9 above)
Comment (kastrubait) [reply rslang/src/app/shared/models/change-storage-type.model.ts]: fix

File: rslang/src/app/shared/models/local-storage-keys.model.ts (hunk 10 above)
Comment (kastrubait) [reply rslang/src/app/shared/models/local-storage-keys.model.ts]: fix

File: rslang/src/app/ebook/components/words-collections/words-collections.component.html (hunk 12 above)
Comment (kastrubait) [reply rslang/src/app/ebook/components/words-collections/words-collections.component.html]: So be it

File: rslang/src/app/ebook/services/ebook-settings-init.service.ts (hunk 14 above)
Comment (kastrubait) [reply rslang/src/app/ebook/services/ebook-settings-init.service.ts]: root means globally. Am I wrong again?

File: rslang/src/app/ebook/services/ebook-settings-init.service.ts (hunk 15 above)
Comment (kastrubait) [reply rslang/src/app/ebook/services/ebook-settings-init.service.ts]: Yes, that's better!!

File: rslang/src/app/ebook/components/ebook-home/ebook-home.component.ts [hunk 16]
```diff
@@ -1,21 +1,67 @@
...
   providers: [WordsDataService],
+  changeDetection: ChangeDetectionStrategy.OnPush,
 })
-export class EbookHome implements OnInit {
+export class EbookHome implements OnInit, OnDestroy {
   wordsCollections: WordsCollection[] = CONFIG_EBOOK.collections;
   words!: Word[];
+  userBookSettings: UserBookSettings;
+  @Input() bookSettingsChanged: UserBookSettings;
+  ebookSettingsSubscription = new Subscription();
```
Comment (Nikitzu) [inline rslang/src/app/ebook/components/ebook-home/ebook-home.component.ts]: Why we create new Subscription if in OnInit we have another one?

File: rslang/src/app/ebook/components/ebook-home/ebook-home.component.ts [hunk 17]
```diff
@@ -1,21 +1,67 @@
...
+  constructor(
+    private ebookSettingsService: EbookSettingsService,
+    private wordsDataService: WordsDataService,
+    private cdr: ChangeDetectorRef,
+    private localStorageService: LocalStorageService,
+  ) {}
 
   ngOnInit(): void {
-    this.words = this.wordsDataService.GetWords();
+    this.ebookSettingsService.firstLoadAndSet();
```
Comment (Nikitzu) [inline rslang/src/app/ebook/components/ebook-home/ebook-home.component.ts]: Method violates Single responsibility princip

File: rslang/src/app/ebook/components/ebook-settings/ebook-settings.component.html [hunk 18]
```diff
@@ -0,0 +1,31 @@
+<section class="ebook-settings">
+  <form #settingsForm="ngForm">
+    <div *ngIf="isUserAuthenticated">
+      <input type="text" id="username" placeholder="Name" />
```
Comment (Nikitzu) [inline rslang/src/app/ebook/components/ebook-settings/ebook-settings.component.html]: Why not mat-form field?

**Review by Nikitzu — CHANGES_REQUESTED** [review]

File: rslang/src/app/ebook/components/ebook-settings/ebook-settings.component.html (hunk 18 above)
Comment (kastrubait) [reply rslang/src/app/ebook/components/ebook-settings/ebook-settings.component.html]: If it is so fundamentally I can change

File: rslang/src/app/ebook/components/ebook-home/ebook-home.component.ts (hunk 17 above)
Comment (kastrubait) [reply rslang/src/app/ebook/components/ebook-home/ebook-home.component.ts]: oh yeah .. fixed, thanks

File: rslang/src/app/shared/services/local-storage.service.ts [hunk 19]
```diff
@@ -35,6 +35,10 @@ export class LocalStorageService {
     return this.storage.key(index);
   }
 
+  isKey(key: string): boolean {
+    return !!(this.storage[key] || '');
```
Comment (Nikitzu) [inline rslang/src/app/shared/services/local-storage.service.ts]: What do we use this method for? Why we dont check keys via hasOwnProperty?

File: rslang/src/app/ebook/services/ebook-settings.service.ts [hunk 20]
```diff
@@ -0,0 +1,26 @@
...
+
+import { LocalStorageService } from '../../shared/services/local-storage.service';
+import { LocalStorageKey } from '../../shared/models/local-storage-keys.model';
+import { EBOOK_SETTINGS } from '../constants/ebook-settings';
+
+@Injectable()
+export class EbookSettingsService {
+  constructor(private localStorageService: LocalStorageService) { }
+
+  firstLoad(): void {
```
Comment (Nikitzu) [inline rslang/src/app/ebook/services/ebook-settings.service.ts]: maybe just load?

File: rslang/src/app/ebook/services/ebook-settings.service.ts [hunk 21]
```diff
@@ -0,0 +1,26 @@
...
+    if (this.localStorageService.isKey(LocalStorageKey.EbookSettings)) {
```
Comment (Nikitzu) [inline rslang/src/app/ebook/services/ebook-settings.service.ts]: why we have method called load but actually it doesn't load anything. It simply checks local storage.

**Review by Nikitzu — CHANGES_REQUESTED** [review]

File: rslang/src/app/ebook/components/ebook-home/ebook-home.component.ts (hunk 16 above)
Comment (kastrubait) [reply rslang/src/app/ebook/components/ebook-home/ebook-home.component.ts]: fix

File: rslang/src/app/ebook/components/ebook-home/ebook-home.component.ts (hunk 11 above)
Comment (kastrubait) [reply rslang/src/app/ebook/components/ebook-home/ebook-home.component.ts]: yes, we can. Fix

File: rslang/src/app/shared/services/local-storage.service.ts (hunk 19 above)
Comment (kastrubait) [reply rslang/src/app/shared/services/local-storage.service.ts]: Ыo be it, thanks for the hint

File: rslang/src/app/ebook/services/ebook-settings.service.ts (hunk 20 above)
Comment (kastrubait) [reply rslang/src/app/ebook/services/ebook-settings.service.ts]: ok

File: rslang/src/app/ebook/services/ebook-settings.service.ts (hunk 21 above)
Comment (kastrubait) [reply rslang/src/app/ebook/services/ebook-settings.service.ts]: Add service for receiving data settings from the server,

File: rslang/src/app/ebook/components/ebook-home/ebook-home.component.ts (hunk 11 above)
Comment (Nikitzu) [reply rslang/src/app/ebook/components/ebook-home/ebook-home.component.ts]: No

File: rslang/src/app/ebook/components/ebook-settings/ebook-settings.component.html (hunk 18 above)
Comment (Nikitzu) [reply rslang/src/app/ebook/components/ebook-settings/ebook-settings.component.html]: Yes please change

File: rslang/src/app/ebook/components/words-collections/words-collections.component.html (hunk 12 above)
Comment (Nikitzu) [reply rslang/src/app/ebook/components/words-collections/words-collections.component.html]: index is not an item

File: rslang/src/app/ebook/services/ebook-settings.service.ts [hunk 22]
```diff
@@ -0,0 +1,49 @@
...
+export class EbookSettingsService {
+  globalSettings: Observable<GlobalSettings>;
+  ebookSettings: UserBookSettings;
+  isUserAuthenticated = false;
+  userId: string;
+
+  constructor(
+    private localStorageService: LocalStorageService,
+    private settingsService: SettingsService,
+  ) { this.globalSettings = this.settingsService.data$; }
```
Comment (Nikitzu) [inline rslang/src/app/ebook/services/ebook-settings.service.ts]: unnecessary variable

**Review by Nikitzu — CHANGES_REQUESTED** [review]

File: rslang/src/app/ebook/services/ebook-settings.service.ts [hunk 23]
```diff
@@ -0,0 +1,47 @@
...
+        this.ebookSettings = data.optionals;
+      });
+      this.localStorageService
+        .setItem(LocalStorageKey.EbookSettings, JSON.stringify(this.ebookSettings));
+    }
+
+    if (Object.prototype.hasOwnProperty.call(localStorage, LocalStorageKey.EbookSettings)) {
+      return;
+    }
+    this.setDefaultSettings(Object.prototype.hasOwnProperty.call(localStorage, LocalStorageKey.EbookSettings));
```
Comment (Nikitzu) [inline rslang/src/app/ebook/services/ebook-settings.service.ts]:
```suggestion
    if (!localStorage.hasOwnProperty(LocalStorageKey.EbookSettings)) {
       this.setDefaultSettings(localStorage.hasOwnProperty(LocalStorageKey.EbookSettings));
    }
```

**Review by Nikitzu — CHANGES_REQUESTED** [review]

File: rslang/.eslintrc.json [hunk 24]
```diff
@@ -26,6 +26,7 @@
         "import/prefer-default-export": "off",
         "class-methods-use-this": "off",
         "lines-between-class-members": "off",
+        "no-prototype-builtins": "warn",
```
Comment (Nikitzu) [inline rslang/.eslintrc.json]: Why warn?

File: rslang/src/app/ebook/components/ebook-settings/ebook-settings.component.html [hunk 25]
```diff
@@ -0,0 +1,32 @@
+<section class="ebook-settings">
+  <form #settingsForm="ngForm">
+    <div *ngIf="isUserAuthenticated">
+      <mat-form-field class="example-full-width">
+        <input matInput id="username" value="User name" />
```
Comment (Nikitzu) [inline rslang/src/app/ebook/components/ebook-settings/ebook-settings.component.html]: Where is the validation?

File: rslang/src/app/ebook/services/ebook-settings.service.ts [hunk 26]
```diff
@@ -0,0 +1,44 @@
...
+        .setItem(LocalStorageKey.EbookSettings, JSON.stringify(this.ebookSettings));
+    }
+
+    if (!localStorage.hasOwnProperty(LocalStorageKey.EbookSettings)) {
+      this.setDefaultSettings();
+    }
+  }
+
+  private setDefaultSettings(): void {
+      const defaultSettings = EBOOK_SETTINGS;
```
Comment (Nikitzu) [inline rslang/src/app/ebook/services/ebook-settings.service.ts]: Why here we have four spaces and int the above method only two? What is the right style?

**Review by Nikitzu — CHANGES_REQUESTED** [review]

File: rslang/.eslintrc.json (hunk 24 above)
Comment (kastrubait) [reply rslang/.eslintrc.json]: Maybe "off", but I think this is a bad idea

File: rslang/src/app/ebook/components/ebook-settings/ebook-settings.component.html (hunk 25 above)
Comment (kastrubait) [reply rslang/src/app/ebook/components/ebook-settings/ebook-settings.component.html]: add

File: rslang/src/app/ebook/services/ebook-settings.service.ts (hunk 26 above)
Comment (kastrubait) [reply rslang/src/app/ebook/services/ebook-settings.service.ts]: I didn't really understand about spaces, There should be two!

File: rslang/src/app/ebook/components/ebook-home/ebook-home.component.ts [hunk 27]
```diff
@@ -1,21 +1,63 @@
...
+import { UserBookSettings } from '../../models/user-book-settings.model';
+import { StorageChanges } from '../../../core/models/change-storage.model';
+import { LocalStorageKey } from '../../../shared/models/local-storage-keys.model';
+import { LocalStorageType } from '../../../shared/models/change-storage-type.model';
+
 @Component({
   selector: 'app-ebook-home',
   templateUrl: './ebook-home.component.html',
   providers: [WordsDataService],
+  changeDetection: ChangeDetectionStrategy.OnPush,
```
Comment (Nikitzu) [inline rslang/src/app/ebook/components/ebook-home/ebook-home.component.ts]: For what?

File: rslang/src/app/ebook/components/ebook-settings/ebook-settings.component.html [hunk 28]
```diff
@@ -0,0 +1,32 @@
...
+        <div *ngIf="isUserAuthenticated">
+            <mat-form-field class="example-full-width">
+                <input matInput id="username" value="User name" formControlName="name" />
+            </mat-form-field>
+        </div>
+        <p class=" subtitle ">Buttons</p>
+        <div *ngFor="let itembtn of userBookSettings.buttonOptions ">
+            <input type="checkbox " id="itembtn.value " name="itembtn.value " [(ngModel)]="itembtn.checked " />
+            <label for="itembtn.value ">
+        {{ itembtn.label }}
```
Comment (Nikitzu) [inline rslang/src/app/ebook/components/ebook-settings/ebook-settings.component.html]: What happened with alignment?

**Review by Nikitzu — CHANGES_REQUESTED** [review]

File: rslang/src/app/ebook/services/ebook-settings.service.ts (hunk 26 above)
Comment (Nikitzu) [reply rslang/src/app/ebook/services/ebook-settings.service.ts]: But there are four.

File: rslang/.eslintrc.json (hunk 24 above)
Comment (Nikitzu) [reply rslang/.eslintrc.json]: Why?

File: rslang/src/app/ebook/components/ebook-home/ebook-home.component.ts (hunk 27 above)
Comment (kastrubait) [reply rslang/src/app/ebook/components/ebook-home/ebook-home.component.ts]: fix

File: rslang/src/app/ebook/components/ebook-settings/ebook-settings.component.html (hunk 28 above)
Comment (kastrubait) [reply rslang/src/app/ebook/components/ebook-settings/ebook-settings.component.html]: fix

File: rslang/.eslintrc.json (hunk 24 above)
Comment (kastrubait) [reply rslang/.eslintrc.json]: I believe that rules should be disabled only in exceptional cases.

**Review by Nikitzu — APPROVED** [review]

**Pull request merged**